cmd/internal/cli/build_linux.go

```diff
@@ -188,6 +188,14 @@ func runBuildRemote(ctx context.Context, cmd *cobra.Command, dst, spec string) {
 		sylog.Fatalf("Unable to build from %s: %v", spec, err)
 	}
 
+	// Ensure that the definition bootstrap source is valid before we submit a remote build
+	if _, err := build.NewConveyorPacker(def); err != nil {
+		sylog.Fatalf("Unable to build from %s: %v", spec, err)
+	}
+	if def.Header["bootstrap"] == "localimage" {
```

Comment:
Might be good to be defensive here?

```suggestion
	if bs, ok := def.Header["bootstrap"]; ok && bs == "localimage" {
```